core/modules/replica/tools/qserv-replica-master-http.cc

```diff
@@ -0,0 +1,380 @@
+    " non-responsive within a configured interval."
+    " The interval is specified via the corresponding command-line option."
+    " And it also has some built-in default value."
+    " Also, note that only a single node failure can trigger the worker"
+    " exclusion sequence."
+    " The controller has the built-in REST API which accepts external commands"
+    " or request for information.";
+
+std::string const logger =
+    "lsst.qserv.replica.qserv-replica-master-http";
+
+class MasterControllerHttpApp
+    :   public Application {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<MasterControllerHttpApp> Ptr;
+
+    /**
+     * The factory method is the only way of creating objects of this class
+     * because of the very base class's inheritance from 'enable_shared_from_this'.
+     *
+     * @param argc
+     *   the number of command-line arguments
+     *
+     * @param argv
+     *   the vector of command-line arguments
+     *
+     * @param description
+     *   the description of the application to be printed when the command-line
+     *   parser is invoked in the "help" mode.
+     *
+     * @param logger
+     *   the name of a logger
+     */
+    static Ptr create(int argc,
+                      const char* const argv[],
+                      std::string const& description,
+                      std::string const& logger) {
+        return Ptr(
+            new MasterControllerHttpApp(
+                argc,
+                argv,
+                description,
+                logger
+            )
+        );
+    }
+
+    // Default construction and copy semantics are prohibited
+
+    MasterControllerHttpApp() = delete;
+    MasterControllerHttpApp(MasterControllerHttpApp const&) = delete;
+    MasterControllerHttpApp& operator=(MasterControllerHttpApp const&) = delete;
+
+    ~MasterControllerHttpApp() override = default;
+
+protected:
+
+    /**
+     * @see MasterControllerHttpApp::create()
+     */
+    MasterControllerHttpApp(int argc,
+                            const char* const argv[],
+                            std::string const& description,
+                            std::string const& logger)
+        :   Application(
+                argc,
+                argv,
+                description,
+                true /* injectDatabaseOptions */,
+                true /* boostProtobufVersionCheck */,
+                true /* enableServiceProvider */
+            ),
+            _healthProbeIntervalSec  (60),
+            _replicationIntervalSec  (60),
+            _workerResponseTimeoutSec(60),
+            _workerEvictTimeoutSec   (3600),
+            _qservSyncTimeoutSec     (60),
+            _numReplicas             (0),
+            _numIter                 (0),
+            _purge          (false),
+            _forceQservSync (false),
+            _permanentDelete(false),
+            _failed         (false),
+            _log(LOG_GET(logger)) {
```

Comment:
I think I mentioned in another file: it would be an improvement to have the defaults in a cfg area or file.

Reply:
💯